lib/buffer.js

```diff
@@ -67,10 +73,67 @@ function Buffer(arg, encoding) {
   return fromObject(arg);
 }
 
+/**
+ * Functionally equivalent to Buffer(arg, encoding) but throws a TypeError
+ * if value is a number.
+ **/
+Buffer.from = function(value, encoding) {
+  if (typeof value === 'number')
+    throw new TypeError(`'value' argument must not be a number`);
+  return typeof value === 'string' ?
+    fromString(value, encoding) :
+    fromObject(value);
+};
+
 Object.setPrototypeOf(Buffer.prototype, Uint8Array.prototype);
 Object.setPrototypeOf(Buffer, Uint8Array);
 
+/**
+ * Creates a new filled Buffer instance.
+ **/
+Buffer.alloc = function(size, fill, encoding) {
+  if (typeof size !== 'number')
+    throw new TypeError(`'size' argument must be a number`);
+  size = Math.max(0, size);
+  if (typeof fill !== 'undefined') {
+    if (encoding) {
```

Comment:
Should we do a strict `if (encoding !== undefined)` so values like `false` can't be passed?